airbyte-ci/connectors/live-tests/src/live_tests/commons/backends/file_backend.py

```diff
@@ -22,101 +30,57 @@ class FileBackend(BaseBackend):
-                for state in states:
-                    states_file.write(f"{state.json()}\n")
-
-    @staticmethod
-    def _get_messages_by_type(messages: Iterable[AirbyteMessage]) -> Dict[str, Union[List, Dict]]:
-        messages_by_type: Dict[str, Union[List, Dict]] = {
-            "catalog": [],
-            "connection_status": [],
-            "records": defaultdict(list),
-            "spec": [],
-            "states": defaultdict(list),
-            "traces": [],
-            "logs": [],
-            "controls": [],
-        }
-
-        for message in messages:
-            if not isinstance(message, AirbyteMessage):
-                continue
-
-            if message.type == AirbyteMessageType.CATALOG:
-                assert isinstance(messages_by_type["catalog"], list)
-                messages_by_type["catalog"].append(message.catalog)
-
-            elif message.type == AirbyteMessageType.CONNECTION_STATUS:
-                assert isinstance(messages_by_type["connection_status"], list)
-                messages_by_type["connection_status"].append(message.connectionStatus)
-
-            elif message.type == AirbyteMessageType.RECORD:
-                assert isinstance(messages_by_type["records"], dict)
-                messages_by_type["records"][message.record.stream].append(message.record)
-
-            elif message.type == AirbyteMessageType.SPEC:
-                assert isinstance(messages_by_type["spec"], list)
-                messages_by_type["spec"].append(message.spec)
-
-            elif message.type == AirbyteMessageType.STATE:
-                if message.state.stream and message.state.stream.stream_descriptor:
-                    stream_name = message.state.stream.stream_descriptor.name
-                    stream_namespace = message.state.stream.stream_descriptor.namespace
-                    key = f"{stream_name}_{stream_namespace}" if stream_namespace else stream_name
-                else:
-                    key = "_global_states"
-                assert isinstance(messages_by_type["states"], dict)
-                messages_by_type["states"][key].append(message.state)
-            elif message.type == AirbyteMessageType.TRACE:
-                assert isinstance(messages_by_type["traces"], list)
-                messages_by_type["traces"].append(message.trace)
-            elif message.type == AirbyteMessageType.LOG:
-                assert isinstance(messages_by_type["logs"], list)
-                messages_by_type["logs"].append(message.log)
-            elif message.type == AirbyteMessageType.CONTROL:
-                assert isinstance(messages_by_type["controls"], list)
-                messages_by_type["controls"].append(message.control)
-        return messages_by_type
+    async def write(self, airbyte_messages: Iterable[AirbyteMessage]):
+        @cached(cache=FileDescriptorLRUCache(maxsize=250))
```

Comment:
I'm not sure I grasp the purpose of the cache here. Can you add a docstring to `write` to explain the implementation details?

Reply:
No problem added a docstring, but I'll reply here too - the cache is to limit the number of files that we have open at any given time. Some connections have many streams (thousands is possible), and we write records & state messages to separate files per stream, so we need a way to avoid having an unbounded number of file descriptors open at once (which could hit up against a limit enforced by the machine running the process). An alternative would be to close the file after every write but for streams with millions of records that is undesirable.